File: app/src/main/java/com/wiseassblog/samsaradayplanner/ApplicationExecutors.java
package com.wiseassblog.samsaradayplanner;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

public class ApplicationExecutors {

    private final Executor background;
    private final Executor mainThread;

    public Executor getBackground() {
        return background;
    }

    public Executor getMainThread() {
        return mainThread;
    }

    public ApplicationExecutors() {
        this.background = Executors.newSingleThreadExecutor();
        this.mainThread = new MainThreadExecutor();
    }

    private static class MainThreadExecutor implements Executor {
        private Handler mainThreadHandler = new Handler(Looper.getMainLooper());

        @Override
        public void execute(Runnable command) {
            mainThreadHandler.post(command);
        }
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ApplicationExecutors.kt
package com.wiseassblog.samsaradayplanner

import android.os.Handler
import android.os.Looper
import java.util.concurrent.Executor
import java.util.concurrent.Executors

class ApplicationExecutors {
    val background: Executor
    val mainThread: Executor

    private class MainThreadExecutor : Executor {
        private val mainThreadHandler = Handler(Looper.getMainLooper())
        override fun execute(command: Runnable) {
            mainThreadHandler.post(command)
        }
    }

    init {
        background = Executors.newSingleThreadExecutor()
        mainThread = MainThreadExecutor()
    }
}